### PR TITLE
Feature/646 restore email reply

### PR DESCRIPTION
**Pull request by inkhey**
related to tracim/tracim#646, tracim/tracim#839, tracim/tracim#762

### CODE REVIEW

File: backend/tracim_backend/lib/mail_fetcher/daemon.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class MailFetcherDaemon(FakeDaemon):
+    """
+    Thread containing a daemon who fetch new mail from a mailbox and
+    send http request to a tracim endpoint to handle them.
+    """
+    def __init__(self, config: 'CFG', burst=True, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.config = config
+        self._fetcher = None  # type: MailFetcher
+        self.burst = burst
```
Comment (buxx) [inline backend/tracim_backend/lib/mail_fetcher/daemon.py]: Can you add doc about burst parameter ? (in MailFetcher class too)

File: backend/tracim_backend/lib/mail_fetcher/daemon.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            host=self.config.EMAIL_REPLY_IMAP_SERVER,
+            port=self.config.EMAIL_REPLY_IMAP_PORT,
+            user=self.config.EMAIL_REPLY_IMAP_USER,
+            password=self.config.EMAIL_REPLY_IMAP_PASSWORD,
+            use_ssl=self.config.EMAIL_REPLY_IMAP_USE_SSL,
+            folder=self.config.EMAIL_REPLY_IMAP_FOLDER,
+            heartbeat=self.config.EMAIL_REPLY_CHECK_HEARTBEAT,
+            use_idle=self.config.EMAIL_REPLY_IMAP_USE_IDLE,
+            connection_max_lifetime=self.config.EMAIL_REPLY_CONNECTION_MAX_LIFETIME,  # nopep8
+            # FIXME - G.M - 2017-11-15 - proper tracim url formatting
```
Comment (buxx) [inline backend/tracim_backend/lib/mail_fetcher/daemon.py]: Remove FIXME (see together)

File: backend/tracim_backend/lib/mail_notifier/notifier.py [hunk 3]
```diff
@@ -125,7 +126,8 @@ def notify_content_update(self, content: Content):
                     self.config,
                     self.session,
                 ).notify_content_update(self._user.user_id, content.content_id)
-        except TypeError as e:
+        except Exception as e:
+            # TODO - G.M - 2018-08-27 - Do Better catching for exception here
             logger.error(self, 'Exception catched during email notification: {}'.format(e.__str__()))
```
Comment (buxx) [inline backend/tracim_backend/lib/mail_notifier/notifier.py]: Add traceback from exception

File: backend/tracim_backend/lib/mail_notifier/notifier.py [hunk 4]
```diff
@@ -125,7 +126,8 @@ def notify_content_update(self, content: Content):
                     self.config,
                     self.session,
                 ).notify_content_update(self._user.user_id, content.content_id)
-        except TypeError as e:
+        except Exception as e:
```
Comment (buxx) [inline backend/tracim_backend/lib/mail_notifier/notifier.py]: After discuss with @lebouquetin, change excepted class or keep.

File: backend/tracim_backend/lib/mail_notifier/notifier.py [hunk 5]
```diff
@@ -129,6 +130,7 @@ def notify_content_update(self, content: Content):
         except Exception as e:
             # TODO - G.M - 2018-08-27 - Do Better catching for exception here
             logger.error(self, 'Exception catched during email notification: {}'.format(e.__str__()))
+            logger.error(self, traceback.format_exc())
```
Comment (buxx) [inline backend/tracim_backend/lib/mail_notifier/notifier.py]: Please use https://docs.python.org/2/library/logging.html#logging.Logger.exception instead, it print traceback automatically.

Comment (inkhey) [reply backend/tracim_backend/lib/mail_notifier/notifier.py]: @buxx  I wanted to do so, but logger use here, is actually not a logging logger, but a specific tracim object from tracim v1 (/lib/utils/logger.py). need to check how to add logger.exception feature in it .

Comment (buxx) [reply backend/tracim_backend/lib/mail_notifier/notifier.py]: @inkhey Okay. I think you can add it with:

``` diff
diff --git a/backend/tracim_backend/lib/utils/logger.py b/backend/tracim_backend/lib/utils/logger.py
index 0b8b4af..6044e01 100644
--- a/backend/tracim_backend/lib/utils/logger.py
+++ b/backend/tracim_backend/lib/utils/logger.py
@@ -48,5 +48,10 @@ class Logger(object):
             Logger.TPL.format(cls=self._txt(instance_or_class), msg=message)
         )
 
+    def exception(self, instance_or_class, message):
+        self._logger.exception(
+            Logger.TPL.format(cls=self._txt(instance_or_class), msg=message)
+        )
+
 
 logger = Logger('tracim')
```

**Pull request merged**